mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulProfileScreenTest.kt

```diff
@@ -44,10 +51,43 @@ class StatefulProfileScreenTest {
       val strings =
           rule.setContentWithLocalizedStrings {
             ProvideFacades(authFacade, socialFacade, chessFacade, speechFacade) {
-              StatefulVisitedProfileScreen("1", {})
+              StatefulVisitedProfileScreen("1", {}, {})
             }
           }
       rule.onNodeWithText(strings.profileMatchTitle("B")).assertExists()
     }
   }
+
+  @Test
+  fun given_userIsLoggedIn_when_clickedOnChallengeFriend_then_prepareGameDialogWithCorrectlySelectedProfile() =
+      runTest {
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(uid = "userId2", name = "user2")) }
+    }
+
+    val authFacade = AuthenticationFacade(auth, store)
+    val chessFacade = ChessFacade(auth, store)
+    val socialFacade = SocialFacade(auth, store)
+    val speechFacade = SpeechFacade(FailingSpeechRecognizerFactory)
+
+    authFacade.signUpWithEmail("user1@email", "user1", "password")
+    val authUser1 = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        socialFacade.profile(uid = "userId2", user = authUser1).filterIsInstance<Profile>().first()
+    authUser1.follow(user2)
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          PawniesTheme {
+            ProvideFacades(authFacade, socialFacade, chessFacade, speechFacade) { Navigation() }
+          }
+        }
+
+    rule.onNodeWithText(strings.sectionSocial).performClick()
+    rule.onNodeWithText("user2").performClick()
+    rule.onNodeWithText(strings.profileChallenge.uppercase()).performClick()
+    rule.onNodeWithText(strings.prepareGamePlay).performClick()
+    rule.onNodeWithText("user2").assertExists()
```

Comment:
I'm not sure this actually tests that the profile is actually selected.

Reply:
Indirectly yes, since It will create a game with the opponent. And if the opponents name is in the screen of the game it is selected in the previous step. This has to be done like that, because I can't test the selected property directly

Reply:
is done